### PR TITLE
Do not update if the model does not change.

### PR DESCRIPTION
**Pull request by boks1971**
Trying to avoid unnecessary updates.

### CODE REVIEW

File: utils/timeseries/timeseries.go [hunk 1]
```diff
@@ -335,7 +335,7 @@ func (t *TimeSeries[T]) ZScore(val T) float64 {
 
 	onlineStdDev := t.onlineStdDevLocked()
 	if onlineStdDev != 0.0 {
-		return (float64(val) - t.welfordM) / t.onlineStdDevLocked()
+		return (float64(val) - t.welfordM) / onlineStdDev
```
Comment (boks1971) [inline utils/timeseries/timeseries.go:338]: random drive-by.

File: utils/protoproxy.go [hunk 2]
```diff
@@ -43,10 +44,11 @@ type ProtoProxy[T proto.Message] struct {
 // this should be used for updates that should be sent periodically, but does not have the urgency of immediate delivery
 // updateFn should provide computations required to generate the protobuf
 // if refreshInterval is 0, then proxy will only update on MarkDirty(true)
-func NewProtoProxy[T proto.Message](refreshInterval time.Duration, updateFn func() T) *ProtoProxy[T] {
+func NewProtoProxy[T proto.Message](refreshInterval time.Duration, updateFn func() T, equalityFn func(T, T) bool) *ProtoProxy[T] {
 	p := &ProtoProxy[T]{
 		updateChan:      make(chan struct{}, 1),
 		updateFn:        updateFn,
+		equalityFn:      equalityFn,
```
Comment (davidzhao) [inline utils/protoproxy.go:51]: is it meaningfully different between types? or is using [proto.Equal](https://pkg.go.dev/google.golang.org/protobuf/proto#Equal) be sufficient?

**Review by davidzhao — APPROVED** [review]
great fix!

File: utils/protoproxy.go (hunk 2 above)
Comment (boks1971) [reply utils/protoproxy.go:51]: I know we are calling this `protoProxy`, but as it is using a generic type, I just gave the flexibility to the instantiating module as it will know about the type. If you think, `proto.Equal` is fine, I can remove the function.

Comment (boks1971) [reply utils/protoproxy.go:51]: oh `T` is just `proto.Message`. Sorry, ok, will remove it.
